---
title: C6312
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C6312
helpviewer_keywords:
- C6312
ms.assetid: 1fc8b9a1-e6ba-4799-84c3-31f289576cca
author: mikeblome
ms.author: mblome
manager: wpickett
ms.workload:
- multiple
ms.openlocfilehash: 5a67173e7621204122a26c2b0ebe632eb01000d0
ms.sourcegitcommit: 94b3a052fb1229c7e7f8804b09c1d403385c7630
ms.translationtype: MT
ms.contentlocale: tr-TR
ms.lasthandoff: 04/23/2019
ms.locfileid: "62781677"
---
# <a name="c6312"></a>C6312
Uyarı C6312: Muhtemel sonsuz döngü: try özel durum filtresi ifadesinde exceptıon_contınue_executıon sabiti kullanın-dışında

 Bu uyarı, sabit kullanımını gösterir. `EXCEPTION_CONTINUE_EXECUTION` (veya -1 olarak değerlendirilen başka bir sabit) yapılandırılmış özel durum işleyicisinin filtre ifadesinde. Sabit değer kullanımını `EXCEPTION_CONTINUE_EXECUTION` sonsuz bir döngüye neden olabilir. Örneğin, donanım tarafından bir özel durum oluştu, özel duruma neden olan yönergeyi yeniden başlatılır. Özel duruma neden olan adresi hala hatalı ise, başka bir özel durum ortaya çıkar ve aynı şekilde ele alınabilir. Bu, sonsuz bir döngüye neden olur.

 Açık bir çağrı `RaiseException` doğrudan sonsuz bir döngüye neden olur, ancak bu kodun protected blokta devam edecektir. Bu beklenmeyen ve, sonsuz bir döngüye neden `RaiseException` geçersiz bir işaretçinin önlemek için kullanıldı.

 Genellikle, `EXCEPTION_CONTINUE_EXECUTION` yalnızca özel duruma neden işaretçi ya da temel alınan bellek düzeltme şansı filtre ifadesinde çağrılan işlev tarafından döndürülen.

## <a name="example"></a>Örnek
 Aşağıdaki kod bu uyarıyı üretir:

```
#include <excpt.h>
#include <stdio.h>
#include <windows.h>

void f (char *ptr)
{
  __try
  {
    // exception occurs if the caller passes null ptr
    // code...
   *ptr = '\0';
  }
  __except (EXCEPTION_CONTINUE_EXECUTION)
    // When EXCEPTION_CONTINUE_EXECUTION is used, the handler
    //  block of the structured exception handler is not executed.
  {
    puts("This block is never executed");
  }
}
```

 Bu uyarıyı düzeltmek için aşağıdaki kodu kullanın:

```
#include <excpt.h>
#include <stdio.h>
#include <windows.h>

void f (char *ptr)
{
  __try
  {
    // exception occurs if the caller passes null ptr
    // code...
    *ptr = '\0';
  }
  __except (GetExceptionCode()== EXCEPTION_ACCESS_VIOLATION ?
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)
  {
    puts("Error Occurred");
  }
}
```